### DISCUSSION TITLE
XLA compilation even further for speed up ⚡️

### DISCUSSION THREAD

**Discussion by sayakpaul** [keras-team/keras-cv#1470] [Ideas]
I am trying to compile the sub-models of Stable Diffusion with XLA in isolation. **_This includes the core diffusion process too_**. See [here](https://github.com/sayakpaul/xla-benchmark-sd/blob/main/exporters/diffusion_model.py#L39-L75).

I am able to obtain a 25% speedup during inference with this. 

However, the text encoder doesn't compile successfully: https://github.com/tensorflow/tensorflow/issues/59818. Do you know anyone that could help? 
 
You can find the compilation details in this repository: https://github.com/sayakpaul/xla-benchmark-sd

@LukeWood @mattdangerw 

Also cc: @deep-diver

**Comment by jbischof** [comment]
@sayakpaul it looks like you're still working with the TF team but let me know if you get stuck.

**Comment by sayakpaul** [comment]
Yeah sure. This is anyway a project for the _keras_ sprint. So, very much line I guess.

**Comment by jbischof** [comment]
I would love to refactor the Stable Diffusion codebase to make it more modular and hackable instead of a monolithic *thing*. Might be worth a discussion down the line!

**Reply by sayakpaul** [reply]
You might want to check out how we (@deep-diver and I) decomposed the Stable Diffusion class into intuitive modules in that regard.

**Reply by jbischof** [reply]
in keras-cv or elsewhere?

**Reply by sayakpaul** [reply]
If you look at https://github.com/sayakpaul/xla-benchmark-sd, you will notice that it decomposes the modules involved in the StableDiffusion class of KerasCV.
